generator/internal/language/rusttemplate.go

```diff
@@ -304,8 +334,15 @@ func rustAnnotateMessage(m *api.Message, state *api.APIState, deserializeWithDef
 		DocLines:          rustFormatDocComments(m.Documentation, state, modulePath, sourceSpecificationPackageName, packageMapping),
 		MessageAttributes: rustMessageAttributes(deserializeWithDefaults),
 		HasNestedTypes:    hasNestedTypes(m),
-		BasicFields: filterSlice(m.Fields, func(s *api.Field) bool {
-			return !s.IsOneOf
+		BasicFields:       basicFields,
+		SingularFields: filterSlice(basicFields, func(f *api.Field) bool {
+			return !isRepeated(f) && !isMap(f)
+		}),
+		RepeatedFields: filterSlice(m.Fields, func(f *api.Field) bool {
+			return isRepeated(f)
+		}),
+		MapFields: filterSlice(m.Fields, func(f *api.Field) bool {
+			return isMap(f)
```

Comment:
Reviewer: check my math, I think this is a partition (no overlaps), but I may have missed something.